lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -239,6 +240,15 @@ def add_customer_data(post, options)
         end
       end
 
+      # created a separate method for these fields because they should not be included
+      # in all transaction types that include methods with source and customer fields
+      def add_extra_customer_data(post, payment_method, options)
+        post[:source][:phone] = {}
+        post[:source][:phone][:number] = options[:phone] || options.dig(:billing_address, :phone)
+        post[:source][:phone][:country_code] = options[:phone_country_code] if options[:phone_country_code]
+        post[:customer][:name] = payment_method.name if payment_method.is_a?(CreditCard) || payment_method.is_a?(NetworkTokenizationCreditCard)
```

Comment:
What would you think about guarding here with `if payment_method.respond_to?(:name)` instead?

Reply:
Lovely, thank you for making me aware of a new Ruby method. I will give it a whirl!